helpers/mysql_stored_procedures.go

```diff
@@ -0,0 +1,148 @@
+        INSERT INTO domains (guid, name, owning_organization_id)
+            VALUES (private_domain_guid, CONCAT(private_domain_name_prefix, private_domain_guid), org_id);
+    END LOOP;
+    CLOSE orgs_cursor;
+END;
+`,
+	},
+	{
+		"assign_user_as_org_manager", `
+CREATE PROCEDURE assign_user_as_org_manager(user_guid TEXT, num_orgs INT)
+BEGIN
+    DECLARE v_user_id INT;
+    DECLARE num_assigned_orgs INT;
+    DECLARE org_id INT;
+    DECLARE org_name_query VARCHAR(255);
+    DECLARE orgs_cursor CURSOR FOR SELECT id FROM organizations WHERE name LIKE org_name_query ORDER BY RAND() LIMIT num_orgs;
+    SET num_assigned_orgs = 0;
+    SET org_name_query = '{{.Prefix}}-org-%';
+
+    SELECT id FROM users WHERE guid = user_guid INTO v_user_id;
+    OPEN orgs_cursor;
+    org_loop: LOOP
+        SET num_assigned_orgs = num_assigned_orgs + 1;
+        IF num_assigned_orgs > num_orgs THEN
+            LEAVE org_loop;
+        END IF;
+        FETCH orgs_cursor INTO org_id;
+        INSERT INTO organizations_managers (organization_id, user_id) VALUES (org_id, v_user_id);
+    END LOOP;
+    CLOSE orgs_cursor;
+END;
+`,
+	},
+	{
+		"create_isolation_segments", `
+CREATE PROCEDURE create_isolation_segments(num_isolation_segments INT)
+BEGIN
+    DECLARE isolation_segment_guid VARCHAR(255);
+	DECLARE	isolation_segment_name_prefix VARCHAR(255);
+    DECLARE counter INT;
+    SET isolation_segment_name_prefix = '{{.Prefix}}-isolation-segment-';
+    SET counter = 0;
+
+    WHILE counter < num_isolation_segments DO
+        SET counter = counter + 1;
+        SET isolation_segment_guid = uuid();
+        INSERT INTO isolation_segments (guid, name)
+            VALUES (isolation_segment_guid, CONCAT(isolation_segment_name_prefix, isolation_segment_guid));
+    END WHILE;
+END;
+`}, {
+		"assign_orgs_to_isolation_segments", `
+CREATE PROCEDURE assign_orgs_to_isolation_segments(num_orgs INT)
+BEGIN
+    DECLARE org_guid VARCHAR(255);
+    DECLARE org_name_query VARCHAR(255);
+    DECLARE isolation_segment_name_query VARCHAR(255);
+    DECLARE v_isolation_segment_guid VARCHAR(255);
+    DECLARE finished INT;
+    DECLARE orgs_cursor CURSOR FOR SELECT guid FROM organizations WHERE name LIKE org_name_query ORDER BY RAND() LIMIT num_orgs;
+    DECLARE CONTINUE HANDLER FOR NOT FOUND SET finished = 1;
+    SET org_name_query = '{{.Prefix}}-org-%';
+    SET isolation_segment_name_query = '{{.Prefix}}-isolation-segment-%';
+    SET finished = 0;
+
+    OPEN orgs_cursor;
+    org_loop: LOOP
+        FETCH orgs_cursor INTO org_guid;
+        IF finished = 1 THEN 
+            LEAVE org_loop;
+        END IF;
+        SELECT guid FROM isolation_segments WHERE name LIKE isolation_segment_name_query ORDER BY RAND() LIMIT 1 INTO v_isolation_segment_guid;
+        INSERT INTO organizations_isolation_segments (organization_guid, isolation_segment_guid)
+            VALUES (org_guid, v_isolation_segment_guid);
+    END LOOP;
+    CLOSE orgs_cursor;
+END;
+`},
+}
```

Comment:
And a final comment on this file: maybe we could adapt the Postgres procedures to use the same (i.e. limited) syntax where available, e.g. `CONCAT` instead of `||` and `WHILE` instead of `FOR`. This would allow us to keep the differences to an absolute minimum.